report_templates/entitlements.erb

```diff
@@ -2,17 +2,28 @@
 name: Subscription - Entitlement Report
 snippet: false
 model: ReportTemplate
+template_inputs:
+- name: Days from Now
+  required: true
+  input_type: user
+  description: "Show subscriptions expiring in <= this many days.  Select 'no limit' to show all entitlements."
+  advanced: false
+  value_type: plain
+  options: "no limit\r\n7\r\n30\r\n60\r\n90\r\n120\r\n180"
 require:
 - plugin: katello
   version: 3.16.0
 -%>
-<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 
-'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 
-'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest' -%>
-<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]]).each_record do |host| -%>
-<%-   host.pools.each do |pool| -%>
+<%- days_from_now = input('Days from Now') -%>
+<%- days_from_now = "" if days_from_now == 'no limit' -%>
+<%- should_filter = days_from_now.present? -%>
+<%- report_headers 'Name', 'Organization', 'Lifecycle Environment', 'Content View', 'Host Collections', 'Virtual', 'Guest of Host', 'OS', 'Arch', 'Sockets', 'RAM', 'Cores', 'SLA', 'Products', 'Subscription Name', 'Subscription Type', 'Subscription Quantity', 'Subscriptions Consumed', 'Subscription SKU', 'Subscription Contract', 'Subscription Account', 'Subscription Start', 'Subscription End', 'Subscription Guest', 'Subscription Expires In' -%>
+<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]], -%>
+<%-   search: should_filter && "pools_expiring_in_days = #{days_from_now}" -%>
+<%- ).each_record do |host| -%>
+<%-   host.pools.expiring_in_days(days_from_now).sort_by { |p| p.end_date }.each do |pool| -%>
```

Comment:
sort_by is actually not allowed in safe mode, getting an error now (not sure how I missed that before), we'll either need to allow that or use 

```
.sort { |a,b| a.end_date <=> b.end_date }
```

Reply:
IMHO sort_by should be allowed

Reply:
@ares @ekohl I added `sort_by` in the Katello PR [here](https://github.com/jeremylenz/katello/blob/bda88d6bf38f5e087e1e186fd5885f2643b61279/app/models/katello/pool.rb#L120) (merged now)